core/thread.go

```diff
@@ -648,3 +630,20 @@ func loadSchema(node *core.IpfsNode, id string) (*schema.Node, error) {
 	}
 	return &sch, nil
 }
+
+// contactUsername returns the username for the peer id if known
+// This is a near-exact port of the Textile version of this method
+func (t *Thread) contactUsername(id string) string {
```

Comment:
We can still put this logic in one place `core/contacts.go`... just make it static and take the `datastore` and `peerId`.

Reply:
(can take `self` too... our own peerId, and current own `username`)

Reply:
yeh that might save it. taking those in a static method was something i tried before, but ran into issues on the mobile side because mobile doesn't have access to the unexported datastore variable... but maybe passing self and username will help me avoid that.